### DISCUSSION TITLE
Get the http referer

### DISCUSSION THREAD

**Discussion by laserir** [zauberzeug/nicegui#1555] [Q&A] [answered]
### Question

Hi folks,
just trying out the login example. When I have some more pages, that all check for the user to be logged in and (if not) opening the /login page, wouldn't it be nice to refer the user automatically back to this page? (How so?)
E.g. if the user is not logged in and opens localhost:8080/page1 (redirecting to /login), after login it should ui.open('/page1')
Any ideas how to automatically achieve this?

**Comment by rodja** [comment, marked as answer]
In #1557 I just created a pull request which uses an url-parameter to tell the login page where to go afterwards. But you could also use `app.storage.user`.

**Reply by rodja** [reply]
I changed the code to use `app.storage.user` to have cleaner urls.
